src/main/java/commission/service/SaleServiceImpl.java

```diff
@@ -27,34 +31,27 @@ public List<Sale> findAll(){
     }
     @Override
     public void insertSale(Sale sale){
-        saleDao.insertSale(sale);
+        saleDao.insertSale(sale, commissionPerSale(sale));
     }
 
+
     @Override
     public Optional<Sale> findSaleById(long id) {
-        var temp = saleDao.findSaleById(id);
-        Sale sale = null;
-        if (temp.isPresent()) {
-            sale = temp.get();
-        } else {
-            throw new ResponseStatusException(NOT_FOUND, "ID not found");
+        try{
+            return saleDao.findSaleById(id);
+        } catch (EmptyResultDataAccessException e){
+            logger.error("ID " + id + " not found");
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
```

Comment:
You can use `HttpStatus.NOT_FOUND` here. `BAD_REQUEST` is abused by a lot of developers but according to the http standard it should be used for malformed payloads. So missing fields in the input json payload, or fields named incorrectly, invalid json etc.

Reply:
Got it! I ve seen the `BAD_REQUEST` in many many places. 
PS: cool im learning good habits